src/ProofTools/DeiterationTool.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Tool to be used during proof mode to perform deiteration on subgraphs on an AEG
+ * @author Dawn Moore
+ */
+
+import {Point} from "../AEG/Point";
+import {AtomNode} from "../AEG/AtomNode";
+import {CutNode} from "../AEG/CutNode";
+import {redrawProof} from "../DrawModes/DrawUtils";
+import {treeContext} from "../treeContext";
+import {illegalColor} from "../Themes";
+import {offset} from "../DrawModes/DragTool";
+import {ProofNode} from "../AEG/ProofNode";
+import {AEGTree} from "../AEG/AEGTree";
+import {highlightChildren} from "../DrawModes/EditModeUtils";
+
+//The node selected with the user mouse down.
+let currentNode: CutNode | AtomNode | null = null;
+
+//Whether or not the node is allowed to be moved (not the sheet).
+let legalNode: boolean;
+
+//The current tree in the proof chain
+let currentProofTree: AEGTree;
+
+/**
+ * Determines the lowest node containing the current point and if that is not the sheet is is
+ * considered a legal node. If it can also deiterate highlight it and all of it's children.
+ * @param event A Mouse up event while using the deiteration tool
+ */
+export function deiterationMouseDown(event: MouseEvent) {
+    const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
+    currentProofTree = new AEGTree(treeContext.getLastProofStep().tree.sheet);
+    currentNode = currentProofTree.getLowestNode(currentPoint);
+
+    isLegal();
+}
+
+/**
+ * Determines the lowest node containing the current point and if that is not the sheet it is
+ * considered a legal node and will be highlighted.
+ * @param event A mouse move event while using deiteration tool
+ */
+export function deiterationMouseMove(event: MouseEvent) {
+    const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
+    currentNode = currentProofTree.getLowestNode(currentPoint);
+    redrawProof();
+
+    isLegal();
+}
+
+/**
+ * If the node we currently have is legal, find it's parent and remove the current node from it.
+ * Push the new version of the tree onto the proof history array.
+ * @param event A mouse up event while using deiteration tool
+ */
+export function deiterationMouseUp(event: MouseEvent) {
+    if (legalNode) {
+        const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
+        if (
+            currentNode !== null &&
+            canDeiterate(currentProofTree.sheet, currentProofTree.getLevel(currentNode))
+        ) {
+            const currentParent: CutNode | null = currentProofTree.getLowestParent(currentPoint);
+            if (currentParent instanceof CutNode) {
+                currentParent.remove(currentPoint);
+            }
+            treeContext.proofHistory.push(new ProofNode(currentProofTree, "Deiteration"));
+        }
+    }
+
+    legalNode = false;
+    redrawProof();
+}
+
+/**
+ * Helper function to determine if the currently selected node is a legal node and highlight it.
+ */
+function isLegal() {

```

Comment:
isLegal makes this sound like a function that returns a bool of whether or not its legal, should be called something more descriptive like "setLegal"